modules/tests/src/main/scala/ch/epfl/scala/debugadapter/testfmk/DebugTestSuite.scala

```diff
@@ -75,7 +78,9 @@ trait DebugTest extends CommonUtils {
     val client = TestingDebugClient.connect(server.uri)
     try {
       server.connect()
-      runAndCheck(client, attach = None, closeSession = true)(steps*)
+      runAndCheck(client, attach = None, closeSession = true, stepFilters = config.stepFilters)(
+        steps*
+      ) // ! That's a bit ugly because we need to pass it to LaunchArguments as well as DebugConfig but LaunchArguments' filters 'overrides' DebugConfig's filters.
```

Comment:
I don't think we need to pass the `stepFilters` to the `runAndCheck` method here. It should already be configured by the `config` via the "build tool".

```suggestion
      runAndCheck(client, attach = None, closeSession = true, stepFilters = config.stepFilters)(
        steps*
      ) // ! That's a bit ugly because we need to pass it to LaunchArguments as well as DebugConfig but LaunchArguments' filters 'overrides' DebugConfig's filters.
```